test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetAddPackageTests.cs

```diff
@@ -0,0 +1,478 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+using NuGet.Test.Utility;
+using NuGet.Versioning;
+using NuGet.XPlat.FuncTest;
+using Xunit;
+
+namespace Dotnet.Integration.Test
+{
+    [Collection("Dotnet Integration Tests")]
+    public class DotnetAddPackageTests
+    {
+        private readonly MsbuildIntegrationTestFixture _fixture;
+
+        public DotnetAddPackageTests(MsbuildIntegrationTestFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        [Fact]
+        public async Task AddPkg_V2LocalSourceFeed_WithRelativePath_NoVersionSpecified_Success()
```

Comment:
Do you need both a V2 and a V3 test case? :)

Reply:
I thought V2 and V3 sources have different directory structure. So I thought testing both of them relative path would be cover all possible scenarios.
I can remove one of them. So we keep V3 one.

Reply:
@nkolev92 
I removed V2 source tests. Please check now.

Reply:
> I thought V2 and V3 sources have different directory structure. So I thought testing both of them relative path would be cover all possible scenarios.

The reason why I kept probing around this topic is the fact that I wanted to make you are focusing on the right scenarios for the test. 

You needed to test 2 things: 

1. `--source` completely overwrites the source.
1. `--source` when used with a relative local source, the full path is calculated correctly.

The full path calculation is not affected by the feed type (v2 or v3 local feed), that only matters afterwards. 

I think the tests are there right now.
Taking another look :)